tests/functional/adapter/test_constraints.py

```diff
@@ -59,18 +59,20 @@ def string_type(self):
     def int_type(self):
         return "INT"
 
+    @pytest.fixture
+    def schema_string_type(self):
```

Comment:
These are really small changes; I don't see an issue with them by themselves. Though this signature doesn't match that of `BaseConstraintsColumnsEqual.schema_string_type()`; I'm not sure if that matters for fixtures.

More generally, have you considered either combining these type fixtures into a single dictionary fixture, which would look similar to `data_types()`, or just making them class attributes? It looks like they are all string literals, and you don't really need a fixture for string literals. I'm trying to limit the amount of test fixtures that we have floating around.

Reply:
> I'm not sure if that matters for fixtures.

I don't think it does - though I'd be open to adding it `string_type` to the signature for sake of readability. I'm just torn whether it actually is more readable given the variable won't be used and that `BaseConstraintsColumnsEqual.schema_string_type()` is defined in a separate repo.. leaning towards leaving it as-is.

> have you considered either combining these type fixtures into a single dictionary fixture, which would look similar to data_types()

I have! The main downside of a single dictionary fixture for the int/string fixtures is that overriding them becomes an all-or-nothing thing where it's more granular currently. For example - the [bigquery](https://github.com/dbt-labs/dbt-bigquery/blob/main/tests/functional/adapter/test_constraints.py#L50-L56) and [snowflake](https://github.com/dbt-labs/dbt-snowflake/blob/main/tests/functional/adapter/test_constraints.py#L34-L40) test implementations just need to override a couple of the fixtures currently, but would need to explicitly write out all the fixture values (at least without some more complicated merging business in the base class). It would be more verbose but I'm generally all for more explicit (even at the cost of duplication) tests.

Reply:
That makes sense. I was thinking it would look something like `self.data_types.update({"int_type": "INT64"}); but in hindsight, I don't know that you can easily reference `self.data_types` if it's a fixture. Maybe if it's a class attribute? I'm also fine leaving as is.